### PR TITLE
Get feat

### PR DESCRIPTION
**Pull request by gildossantos**
Main change is to flesh out the Insertion object, but also some renames of functions and clarification of stuff in docs.

### CODE REVIEW

File: harvdev_utils/psycopg_functions/fb_feature_classes.py [hunk 1]
```diff
@@ -498,10 +498,138 @@ class Insertion(Feature):
...
+        if type(self.floc_list) is None:
+            log.warning('The "floc_list" is None - expected a list (even if empty).')
+        elif type(self.floc_list) != list:
+            log.warning('The "floc_list" is not a list - expected a list (even if empty).')
+        elif len(self.floc_list) == 0:
+            log.debug('Can\'t determine feature location for {}: no featureloc info in chado.'.format(self.uniquename))
+        elif len(self.floc_list) > 1:
+            log.debug('Can\'t determine feature location for {}: many featureloc values in chado.'.format(self.uniquename))
+        else:
+            self.chr_id = self.floc_list[0][0]
```
Comment (ianlongden) [inline harvdev_utils/psycopg_functions/fb_feature_classes.py:540]: Being picky I know. But can we have these 0-4 defined as constants and referenced here:-
self.chr_id = self.floc_list[0][CHR_ID] (or something) for all these.

File: harvdev_utils/psycopg_functions/fb_feature_classes.py [hunk 2]
```diff
@@ -498,10 +498,138 @@ class Insertion(Feature):
...
+            self.strand = self.floc_list[0][3]
+            if type(self.fbrf_list) == list and re.match(r'^FBrf[0-9]{7}$', str(self.floc_list[0][4])):
+                self.fbrf_list.append(self.floc_list[0][4])
+
+        return
+
+    def get_agr_floc(self):
+        """Determine onbase start/stop coordinates and from that, distinguish insertion from delin."""
+        # First make sure necessary floc info is present.
+        if self.fmin is None:
```
Comment (ianlongden) [inline harvdev_utils/psycopg_functions/fb_feature_classes.py:552]: Thats a lot of if else, would be nice to have as function calls i.e.

def teis():
   do something

def is():
   do something else

process_dict = {
   'transposable_element_insertion_site': teis,
   'insertion_site': is
}

if bob in process_dict:
    process_dict[bob]()

Does that make sense ?

Comment (ianlongden) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:552]: Sorry having problems with the formatting..... grrr.

Comment (gildossantos) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:552]: Great suggestion. Thanks!

File: harvdev_utils/psycopg_functions/fb_feature_classes.py (hunk 1 above)
Comment (gildossantos) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:540]: Yes, I see your point. I'm using a generic function that just adds sql query results (via psycopg2) to my objects as a list of tuples. I'll look into ways that I can apply labels for each tuple position, and turn it into a dict (key:value pairs). Does that seem ok?

Comment (ianlongden) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:540]: What I meant was to have something like 
CHR_ID = 0
and use that later which would be easier but there are other solutions.

Comment (gildossantos) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:540]: oh, so just within this method, specify CHR_ID = 0, FMIN = 1, etc first, then use those constants in the final "else" step. have I got it now?

Comment (ianlongden) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:540]: Yep, bingo you success.

Comment (christabone) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:540]: @ianlongden does this with some of the Python scrips and it makes it much readable when you look back at the code. I concur.

Comment (gildossantos) [reply harvdev_utils/psycopg_functions/fb_feature_classes.py:540]: For sure - already changed in my local branch

**Pull request merged**